ql/src/java/org/apache/hadoop/hive/ql/Compiler.java

```diff
@@ -265,7 +266,15 @@ private void openTransaction(TxnType txnType) throws LockException, CommandProce
     if (DriverUtils.checkConcurrency(driverContext) && startImplicitTxn(driverContext.getTxnManager()) &&
         !driverContext.getTxnManager().isTxnOpen() && txnType != TxnType.COMPACTION) {
       String userFromUGI = DriverUtils.getUserFromUGI(driverContext);
-      driverContext.getTxnManager().openTxn(context, userFromUGI, txnType);
+      boolean isHiveReplTxn = (driverContext.getQueryState().getHiveOperation() != null)
+              && (driverContext.getQueryState().getHiveOperation().equals(HiveOperation.REPLDUMP)
+              || driverContext.getQueryState().getHiveOperation().equals(HiveOperation.REPLLOAD));
+      String dbUnderReplication = (isHiveReplTxn) ? PlanUtils.stripQuotes(tree.getChild(0).getText()) : null;
+      if (dbUnderReplication != null) {
```

Comment:
Can isHiveReplTxn be true and dbUnderReplication be null ever? if not, this should be based on check isHiveReplTxn

Reply:
Done